nevergrad/parametrization/core.py

```diff
@@ -263,6 +263,23 @@ def set_name(self: P, name: str) -> P:
 
     # %% Constraint management
 
+    def constraint_violation(self) -> float:
+        """Returns maximum of constraint values.u
+        Uses 1 for True, constraint violation for float, negative numbers ok
+        when inside the satisfaction domain.
```

Comment:
why do you use a max, a wouldnt a sum provide more information?
Also I don't understand the docstring at all, can you clarify ? negative values are violation as far as I understand, so not OK

Reply:
Yes a sum provides more information. However, here, it would not work.

We want something which solves all the constraints, not something which solves the sum of the constraints.
This is just not equivalent.

We should use the exact value of each constraint for learning a model of constraints, and use it: for the moment this is not included in Nevergrad. This is the main next step. This is not for the current PR, and using the sum as a substitute would be a bad idea.

I improve the docstring.

Reply:
I also add an explanation of the critical "tell".